ros/node.go

```diff
@@ -41,21 +40,13 @@ type defaultNode struct {
 	waitGroup      sync.WaitGroup
 }
 
-func listenRandomPort(address string, trialLimit int) (net.Listener, error) {
-	var listener net.Listener
-	var err error
-	numTrial := 0
-	for numTrial < trialLimit {
-		port := 1024 + rand.Intn(65535-1024)
-		addr := fmt.Sprintf("%s:%d", address, port)
-		listener, err = net.Listen("tcp", addr)
-		if err == nil {
-			return listener, nil
-		} else {
-			numTrial += 1
-		}
+func listenRandomPort(address string) (net.Listener, error) {
+	addr := fmt.Sprintf("%s:0", address)
```

Comment:
I suppose it would be better to write here a comment about this behavior.
I'm not sure but it can be specific only [for linux](https://github.com/torvalds/linux/blob/master/net/ipv4/inet_connection_sock.c#L275).
So, my suggestion is to write something here and in README, to make everyone know about this restriction.

Reply:
> I'm not sure but it can be specific only for linux.

It's not. Here - https://www.dnorth.net/2012/03/17/the-port-0-trick/
That's a very old note, but I suppose still the same behavior in modern OS.